app/concerns/agent_controller_concern.rb

```diff
@@ -68,7 +68,8 @@ def control!
             log "Agent '#{target.name}' is disabled"
           end
         when 'configure'
-          target.update! options: target.options.deep_merge(interpolated['configure_options'])
+          new_options = target.options.deep_merge(interpolated['configure_options']) { |key, old, new| !(new.kind_of?(Array) && old.kind_of?(Array)) ? new : (old+new).uniq }
```

Comment:
It only runs the new "merge arrays" block when both `new` and `old` are arrays. Otherwise, it simply sets the value to `new`, like it would by default(without the addition of this block).